src/app/containers/Image/index.jsx

```diff
@@ -27,14 +27,15 @@ const getCopyright = copyrightHolder => {
 const shouldLazyLoad = position =>
   !!urlWithPageAnchor() || position[0] > LAZYLOAD_FROM_BLOCK;
 
-const ImageContainer = ({ blocks, position }) => {
+const ImageContainer = ({ blocks, position, shouldPreload }) => {
   if (!blocks) {
     return null;
   }
 
   const rawImageBlock = filterForBlockType(blocks, 'rawImage');
   const altTextBlock = filterForBlockType(blocks, 'altText');
   const captionBlock = filterForBlockType(blocks, 'caption');
+  const ShouldPreLoadLeadImage = position[0] <= 4 && shouldPreload;
```

Comment:
```suggestion
  const ShouldPreLoadLeadImage = position[0] < LAZYLOAD_FROM_BLOCK && shouldPreload;
```
Where does the requirement come from the preload if the image appears in the 4th position or below? I'm just wondering if it should be aligned with the constant used to decide when to lazy load an image, I don't imagine you want to both preload and lazy load an image?

Reply:
I think we need tests for the application of this logic covering when the `preload` prop should be true or false

Reply:
Is there a way to unit test this within the image container as the logic to add the link tag to the head is within `ImageWithPlaceholder` wouldn't the unit test for that cover the logic? As the only thing I could test directly on the `Image` container is to check the prop `preload` but I don't think that's possible with Reacts testing library

Reply:
I was thinking more put some tests together that:
- Pass in an image with postion set to 2 -> a preload link tag is added to the head
- Pass in an image with postion set to 5 -> no preload link tag is added to the head

Just testing that logic that determines whether an image should be preloaded, hope that makes sense, slack me for a call if not happy to go through it :)

On an unrelated note, suggestion may not be correct I just wanted to know if that came from product or was an engieers choice? I guess this must be a roundabout way of determining which image is the hero image?

Reply:
Cool, I have written a test for that. The reason I'm questioning is because technically writing a test like that, is testing both the logic for the images block position in `Image` and the logic for adding the preload link tag to the head in `ImageWithPlaceholder` which couples them together.

I'll find someone who might know about that choice, but I think you're right, if the image is before the threshold for lazy loading it's determined to be the hero image.

Reply:
> Cool, I have written a test for that. The reason I'm questioning is because technically writing a test like that, is testing both the logic for the images block position in Image and the logic for adding the preload link tag to the head in ImageWithPlaceholder which couples them together.

Oh I see what you mean, if you've written it already give me a shout when it's pushed and I'll see if we can get rid of some duplication. In this codebase we tend to not mock out dependencies much so we have a degree of duplication going on through that decision. To avoid duplication, you could mock out the `ArticleFigure` component and just assert whether the `preload` prop has been set to true or false; I sometimes think though that couples your test to your code a bit too much. Anyway, if you have a test written we can discuss and see if we get happy medium :)